### PR TITLE
fix: fixed update order status

### PR DESCRIPTION
**Pull request by dmytroivasikiv**

### CODE REVIEW

File: src/provider/foodora/foodora.service.ts [hunk 1]
```diff
@@ -390,22 +389,33 @@ export class FoodoraService implements ProviderService {
...
     try {
       if (orderData.orderStatus === OrderStatusEnum.COMPLETED) {
-        await this.markFoodoraOrderAsPrepared(orderId);
+        await this.markFoodoraOrderAsPrepared(existingOrder.orderId);
       } else if (orderData.orderStatus === OrderStatusEnum.IN_PROGRESS) {
-        await this.updateFoodoraOrderStatus(orderId, {
+        await this.updateFoodoraOrderStatus(existingOrder.orderId, {
           acceptanceTime: new Date().toISOString(),
-          remoteOrderId: orderId.split('-_-')[1],
+          remoteOrderId: existingOrder.orderId.split('-_-')[1],
```
Comment (vicnguyenqt) [inline src/provider/foodora/foodora.service.ts:405]: What does this `split('-_-')` do

File: src/provider/foodora/foodora.service.ts [hunk 2]
```diff
@@ -420,7 +430,7 @@ export class FoodoraService implements ProviderService {
       throw new HttpException('Failed to update Foodora order', HttpStatus.BAD_REQUEST);
     }
 
-    const order = await this.getOrderDetails(orderId.split('-_-')[1]);
+    const order = await this.getOrderDetails(existingOrder.orderId.split('-_-')[1]);
```
Comment (vicnguyenqt) [inline src/provider/foodora/foodora.service.ts:433]: This `-_-` now is repeatedly, consider convert into shared function.

File: src/provider/provider-management.service.ts [hunk 3]
```diff
@@ -166,22 +166,21 @@ export class ProviderManagmentService {
 
     //Check business from order data
 
-    const order = await this.getOrderById(orderId, orderingUserId);
+    const order = await this.getOrderById(orderId, orderingUserId);
```
Comment (vicnguyenqt) [inline src/provider/provider-management.service.ts:169]: why not just 
```suggestion
    const { business } = await this.getOrderById(orderId, orderingUserId);    
```

**Review by vicnguyenqt — COMMENTED** [review]
Small notes, note sure the PR is about. Consider to change

**Review by quochuy278 — APPROVED** [review]

File: src/provider/foodora/foodora.service.ts (hunk 1 above)
Comment (dmytroivasikiv) [reply src/provider/foodora/foodora.service.ts:405]: The order id is a part of the token, in the format “UE9fRkktZXdkYw-_-oma_a2461d0a-0688-5652-a6b5-2272c3f72c31-_-eu”, where the central part is the id

Comment (vicnguyenqt) [reply src/provider/foodora/foodora.service.ts:405]: Yup, so let's move this into shared function instead. Something like: `getOrderIdFoodora(orderId)`

File: src/provider/provider-management.service.ts (hunk 3 above)
Comment (dmytroivasikiv) [reply src/provider/provider-management.service.ts:169]: the line with const `{ business } = order;` is generally not needed, it has an error and is not used. I donʼt understand why it is there but I didnʼt write this logic so I left it

Comment (vicnguyenqt) [reply src/provider/provider-management.service.ts:169]: Ok 👍 .  @quochuy278  :?

**Pull request merged**